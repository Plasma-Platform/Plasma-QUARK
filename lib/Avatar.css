.avatar{vertical-align:middle}.avatar_round{border-radius:50%}