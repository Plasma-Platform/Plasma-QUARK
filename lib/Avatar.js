'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _md = require('md5');

var _md2 = _interopRequireDefault(_md);

var _isRetina = require('is-retina');

var _isRetina2 = _interopRequireDefault(_isRetina);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Avatar = function (_React$Component) {
  (0, _inherits3.default)(Avatar, _React$Component);

  function Avatar(props) {
    (0, _classCallCheck3.default)(this, Avatar);

    var _this = (0, _possibleConstructorReturn3.default)(this, (Avatar.__proto__ || (0, _getPrototypeOf2.default)(Avatar)).call(this, props));

    _this.state = {
      avatarSrc: _this.props.src || _this.getGravatarUrl(),
      isAvatarLoadError: false
    };


    _this.handleAvatarLoadError = _this.handleAvatarLoadError.bind(_this);

    _this.bgColors = ['#1a76d2', '#546e7a', '#e64a19', '#0b8738', '#ffa001', '#996969', '#42a5f5', '#243238', '#006023', '#ff6f00'];

    _this.avatarClassName = _this.props.className ? 'avatar ' + _this.props.className : 'avatar';
    return _this;
  }

  (0, _createClass3.default)(Avatar, [{
    key: 'getGravatarUrl',
    value: function getGravatarUrl() {
      return '//www.gravatar.com/avatar/' + (0, _md2.default)(this.props.email) + '?s=' + ((0, _isRetina2.default)() ? this.props.size * 2 : this.props.size) + '&d=404';
    }
  }, {
    key: 'handleAvatarLoadError',
    value: function handleAvatarLoadError() {
      if (this.state.avatarSrc === this.props.src && this.props.email.length > 0) {
        this.setState({
          avatarSrc: this.getGravatarUrl(),
          isAvatarLoadError: false
        });
      } else if (this.state.avatarSrc === this.getGravatarUrl()) {
        this.setState({
          avatarSrc: '',
          isAvatarLoadError: true
        });
      }
    }
  }, {
    key: 'renderAvatarImgByInitials',
    value: function renderAvatarImgByInitials() {
      var nameParts = this.props.name.split(' ');
      var emailParts = this.props.email.split('');

      var bgColor = this.bgColors[Math.floor(Math.random() * 10 + 1)];

      var initials = void 0;

      if (nameParts.length > 1 && nameParts[0].length === 1 && nameParts[1].length === 1) {
        initials = '' + nameParts[0] + nameParts[1];
      } else if (emailParts.length > 1 && emailParts[0].length === 1 && emailParts[1].length === 1) {
        initials = '' + emailParts[0] + emailParts[1];
      } else {
        initials = 'N/A';
      }

      return _react2.default.createElement(
        'svg',
        {
          className: this.avatarClassName,
          height: this.props.size,
          width: this.props.size,
          viewBox: '0 0 ' + this.props.size + ' ' + this.props.size
        },
        _react2.default.createElement('rect', {
          width: this.props.size,
          height: this.props.size,
          stroke: bgColor,
          'stroke-width': '0',
          fill: bgColor
        }),
        _react2.default.createElement(
          'text',
          {
            x: this.props.size / 2,
            y: this.props.size / 2,
            fill: '#fff',
            fontSize: this.props.size / 3,
            fontFamily: 'PT Sans, sans-serif',
            textAnchor: 'middle',
            dominantBaseline: 'central'
          },
          initials.toUpperCase()
        )
      );
    }
  }, {
    key: 'renderAvatarImgBySrc',
    value: function renderAvatarImgBySrc() {
      return _react2.default.createElement('img', {
        className: this.avatarClassName,
        src: this.state.avatarSrc,
        alt: this.props.name.length > 0 ? this.props.name : this.props.email,
        onError: this.handleAvatarLoadError
      });
    }
  }, {
    key: 'render',
    value: function render() {
      return this.state.isAvatarLoadError ? this.renderAvatarImgByInitials() : this.renderAvatarImgBySrc();
    }
  }]);
  return Avatar;
}(_react2.default.Component);

Avatar.propTypes = {
  src: _react2.default.PropTypes.string,
  email: _react2.default.PropTypes.string,
  name: _react2.default.PropTypes.string,
  size: _react2.default.PropTypes.number,
  className: _react2.default.PropTypes.string
};
Avatar.defaultProps = {
  size: 60,
  name: '',
  email: ''
};
exports.default = Avatar;