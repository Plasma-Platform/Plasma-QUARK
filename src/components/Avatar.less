.avatar {
  vertical-align: middle;

  &_round {
    border-radius: 50%;
  }
}
