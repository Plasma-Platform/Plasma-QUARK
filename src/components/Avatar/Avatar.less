.tm-quark-avatar {
  vertical-align: middle;
}
