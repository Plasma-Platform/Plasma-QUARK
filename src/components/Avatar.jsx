import React from 'react';
import md5 from 'md5';
import fetchJSONP from 'fetch-jsonp';
import isRetina from 'is-retina';
import Base64 from 'js-base64';
import Underscore from 'underscore';
import './Avatar.less';

export default class Avatar extends React.Component {
  static PropTypes = {
    email     : React.PropTypes.string,
    src       : React.PropTypes.string,
    name      : React.PropTypes.string,
    size      : React.PropTypes.number,
    className : React.PropTypes.string,
    rating    : React.PropTypes.string
  };

  static defaultProps = {
    size      : 60,
    rating    : 'g',
    className : '',
    src       : '',
    name      : ''
  };

  constructor (props) {
    super(props);
    this.componentIsMounted = false;
    this.url = 'https://gravatar.com/';
    this.default = 'https://cdn2wp-templatemonster.netdna-ssl.com/wp-content/uploads/2016/10/onepixel.png';
    this.colors = [ // array colors
      '#1A76D2',
      '#546E7A',
      '#E64A19',
      '#0B8738',
      '#FFA001',
      '#996969',
      '#42A5F5',
      '#243238',
      '#006023',
      '#FF6F00'
    ];
    this.state = {
      status      : null,
      color       : '',
      initial     : '',
      displayName : '',
      size        : this.props.size,
      avatar      : ''
    };
  }

  /**
   * Main function
   *
   * @param {Object} atts
   * @returns {Promise.<T>}
   */
  getProfile = Underscore.memoize((atts = {}) => {
    return fetchJSONP(`${this.url}${md5(atts.email)}.json`).then((response) => {
      return response.json();
    }).then((response) => {
      let data = response.entry[0];
      return {
        status      : 1,
        color       : this.getColor(data.displayName),
        initial     : this.prepareInitials(data.displayName),
        displayName : data.displayName,
        size        : this.getSize(atts.size),
        avatar      : this.getGravatarSrc(data.thumbnailUrl, atts)
      };
    }).catch(() => {
      return {
        status : 0
      };
    });
  }, (input) => {
    return JSON.stringify(input);
  });

  /**
   * Generate size for retina display
   *
   * @param {number} size
   * @returns {number}
   * @private
   */
  getSize = (size) => {
    let modernBrowser = true;  // server-side, we render for modern browsers
    if (typeof window !== 'undefined') {
      // this is not NodeJS
      modernBrowser = 'srcset' in document.createElement('img');
    }
    return !modernBrowser && isRetina() ? size * 2 : size;
  };

  /**
   * Generate src for gravatar with query params
   *
   * @param {String} cleanURL
   * @param {Object} atts
   * @returns {string}
   * @private
   */
  getGravatarSrc = (cleanURL, atts) => {
    const query = `a=${this.getSize(atts.size)}&r=${atts.rating}&d=${this.default}`;
    return `${cleanURL}?${query}`;
  };

  /**
   * Get initials from gravatar display name
   *
   * @param {String} name
   * @returns {string}
   * @private
   */
  prepareInitials = (name) => {
    if (name) {
      const parts = name.split(' ');
      let result = '';
      if (parts.length > 1) {
        result = (parts[0][0] + parts[1][0]).toUpperCase();
      } else if (parts[0].length > 1) {
        result = (parts[0][0] + parts[0][1]).toUpperCase();
      }
      return result;
    }
  };

  /**
   * Check the color for display name
   *
   * @param {String} name
   * @returns {string}
   * @private
   */
  getColor = (name = '') => {
    const nameExist = !!name;
    if (nameExist) {
      const literal = name[0].toUpperCase();
      return this.colors[literal.charCodeAt(0).toString()[0]];
    } else {
      return this.colors[0];
    }
  };

  /**
   * Returns svg avatar based on name provided
   * @param {string} name
   * @param {string} bg
   * @returns {string}
   * @private
   */
  getTextAvatar = (name, bg = '#1A76D2') => {
    const SVG = document.createElement('svg');
    SVG.setAttribute('xmlns', 'http://www.w3.org/2000/svg');
    SVG.setAttribute('pointer-events', 'none');
    SVG.setAttribute('width', this.state.size);
    SVG.setAttribute('height', this.state.size);
    SVG.setAttribute('style', 'background-color: ' + bg);

    const TEXT = document.createElement('text');
    TEXT.setAttribute('text-anchor', 'middle');
    TEXT.setAttribute('y', '50%');
    TEXT.setAttribute('x', '50%');
    TEXT.setAttribute('dy', '0.35em');
    TEXT.setAttribute('style', `font-family: "PT_Sans",sans-serif; font-size: ${this.props.size * 0.3}px`);
    TEXT.setAttribute('pointer-events', 'auto');
    TEXT.setAttribute('fill', 'white');

    TEXT.innerHTML = name;

    SVG.appendChild(TEXT);
    const svgString = SVG.outerHTML.toString();
    return `data:image/svg+xml;base64,${Base64.Base64.encodeURI(svgString)}`;
  };

  getSimpleImageBlock = (src = '') => {
    return (
      <div className={`avatar ${this.props.className}`}>
        <img
          src={src}
          height={this.state.size}
          width={this.state.size}
          alt="image"
        />
      </div>
    );
  };

  getGravatarBlock = () => {
    return (
      <div className={`avatar ${this.props.className}`}>
        <img
          alt="initials"
          height={this.state.size}
          width={this.state.size}
          src={this.getTextAvatar(this.state.initial, this.state.color)}
          className="avatar__initials"
        />
        <img
          alt="avatar"
          height={this.state.size}
          width={this.state.size}
          src={this.state.avatar}
          className="avatar__image"
        />
      </div>
    );
  };

  setGravatarInfo = (props) => {
    if (!props.src && !!props.email) {
      this.getProfile(props).then(data => {
        if (this.componentIsMounted) {
          this.setState(data);
        }
      });
    }
  };

  componentWillUnmount () {
    this.componentIsMounted = false;
  }

  componentWillReceiveProps (newProps) {
    this.setGravatarInfo(newProps);
  }

  componentDidMount () {
    this.componentIsMounted = true;
    this.setGravatarInfo(this.props);
  }

  render () {
    let html = '';
    const srcExist = !!this.props.src;
    if (srcExist) {
      html = this.getSimpleImageBlock(this.props.src);
    } else if (this.state.status) {
      html = this.getGravatarBlock();
    } else {
      const nameExist = !!this.props.name;
      const emailExist = !!this.props.email;
      if ((!nameExist && !emailExist) || (emailExist && this.state.status === null)) {
        return false;
      }
      const initials = this.prepareInitials(nameExist ? this.props.name : this.props.email);
      const color = this.getColor(initials);
      html = this.getSimpleImageBlock(this.getTextAvatar(initials, color));
    }
    return html;
  }

}
