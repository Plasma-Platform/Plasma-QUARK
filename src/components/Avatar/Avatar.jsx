import React, { Component } from 'react';
import PropTypes from 'prop-types';
import md5 from 'md5';
import isRetina from 'is-retina';
import ColorHash from 'color-hash';
import NearestColor from 'nearest-color';

import './Avatar.less';

const BG_COLORS = {
  1: '#1a76d2',
  2: '#546e7a',
  3: '#e64a19',
  4: '#0b8738',
  5: '#ffa001',
  6: '#996969',
  7: '#42a5f5',
  8: '#243238',
  9: '#006023',
  10: '#ff6f00',
};

export default class Avatar extends Component {
  static propTypes = {
    src: PropTypes.string,
    email: PropTypes.string,
    name: PropTypes.string,
    size: PropTypes.number,
    className: PropTypes.string,
  }

  static defaultProps = {
    src: '',
    email: '',
    name: '',
    size: 60,
    className: '',
  }

  state = {
    isSrcError: false,
  }

  getGravatarRequestUrl = (email) => {
    if (!email) {
      return null;
    }

    const { size } = this.props;

    return `//www.gravatar.com/avatar/${md5(email)}?s=${isRetina()
      ? size * 2
      : size
    }&d=404`;
  }

  handleImageLoadError = () => {
    this.setState(() => ({
      isSrcError: true,
    }));
  }

  renderImageByInitials() {
    const {
      name = '',
      email = '',
      size,
    } = this.props;

    const nameParts = name.trim().replace(/\s{2,}/g, ' ').split(' ');
    const emailParts = email.trim().split('');

    const initialsByFirstName = (
      nameParts.length >= 1
      && nameParts[0].split('').length > 1
    )
      ? `${nameParts[0].split('')[0]}${nameParts[0].split('')[1]}`
      : null;

    const initialsByName = (
      nameParts.length >= 2
      && nameParts[0].split('').length > 0
      && nameParts[1].split('').length > 0
    )
      ? `${nameParts[0].split('')[0]}${nameParts[1].split('')[0]}`
      : null;

    const initialsByEmail = emailParts.length >= 2
      ? `${emailParts[0]}${emailParts[1]}`
      : null;

    const initials = (initialsByName || initialsByFirstName || initialsByEmail || 'N/A').toUpperCase();

    const colorHash = new ColorHash();
    const initialsColor = colorHash.hex(initials);
    const nearestColor = NearestColor.from(BG_COLORS);
    const imageBg = nearestColor(initialsColor).value;

    this.imageUrl = null;

    return (
      <svg
        className={`tm-quark-avatar ${this.props.className}`}
        height={size}
        width={size}
      >
        <circle
          cx={size / 2}
          cy={size / 2}
          r={size / 2}
          fill={imageBg}
        />

        <text
          x={size / 2}
          y={size / 2}
          fill="#fff"
          fontSize={size / 3}
          textAnchor="middle"
          dy=".3em"
        >
          {initials}
        </text>
      </svg>
    );
  }

  renderImageByURL(url) {
    const {
      src,
      email,
      name,
      className,
      size,
      ...props
    } = this.props;

    this.imageUrl = url;

    return (
      <img
        className={`tm-quark-avatar ${this.props.className}`}
        width={size}
        height={size}
        src={url}
        alt={name || email || ''}
        onError={this.handleImageLoadError}
        {...props}
      />
    );
  }

  render() {
    const {
      src,
      email,
      name,
    } = this.props;

    const gravatarRequestUrl = this.getGravatarRequestUrl(email);
    const gravatarUrl = gravatarRequestUrl && !(
      this.state.isSrcError && this.imageUrl === gravatarRequestUrl
    )
      ? gravatarRequestUrl
      : null;
    const srcUrl = src && !(this.state.isSrcError && this.imageUrl === src)
      ? src
      : null;
    const imageUrl = srcUrl || gravatarUrl;

    if (!imageUrl) {
      return this.renderImageByInitials({
        name,
        email,
      });
    }

    return this.renderImageByURL(imageUrl);
  }
}
