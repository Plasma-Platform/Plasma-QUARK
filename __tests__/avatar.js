/* global expect:false, test:false, it:false, describe:false, beforeEach:false, afterEach:false */

import React from 'react';
import ReactDOM from 'react-dom';
import {mount} from 'enzyme';
import Avatar from '../src/components/Avatar';
let avatar;
let params = {
  email  : 'stevenreed@templatemonster.me',
  size   : 60,
  name   : 'Tester',
  rating : 'g'
};
let container;
beforeEach(() => {
  container = document.createElement('div');
  avatar = mount(<Avatar email="shaggrath@mail.ru" />, {attachTo: container});
});

afterEach(() => {
  ReactDOM.unmountComponentAtNode(container);
});

describe('Functional methods test:', () => {
  it('Method: getProfile(), Promise is work normal', async function() {
    try {
      const data = await avatar.instance().getProfile(params);
      expect(data.status).toEqual(1);
    } catch (object) {
      throw object;
    }
  });

  it('Method: getProfile(), Promise is work normal with unreal email', async function() {
    let localParams = Object.assign({}, params);
    localParams.email = 'fackeEmail@faceDomain.com';
    try {
      const data = await avatar.instance().getProfile(localParams);
      expect(data.status).toEqual(0);
    } catch (object) {
      throw object;
    }
  });

  it(`Method: getSize(), must return size (number) from ${params.size}`, () => {
    expect(typeof avatar.instance().getSize(params.size)).toEqual('number');
  });

  it(`Method: setGravatarInfo(), must return url (string)`, async function() {
    try {
      await avatar.instance().setGravatarInfo(params);
    } catch (object) {
      throw object;
    }
  });

  describe(`Method: prepareInitials(), must return string with length 2`, () => {
    const phrases = ['Hello', 'Hello world', 'Hello my world'];
    phrases.forEach((element) => {
      it(`Call method with phrase "${element}" return the first two letters`, () => {
        expect(avatar.instance().prepareInitials(element).length).toEqual(2);
        expect(typeof avatar.instance().prepareInitials(element)).toEqual('string');
      });
    });
  });

  describe(`Method: getColor(), must return HEX color code`, () => {
    const testData = [
      {
        name  : 'SR',
        color : '#006023'
      },
      {
        name  : 'ะกะก',
        color : '#546e7a'
      },
      {
        name  : '',
        color : '#1a76d2'
      }
    ];
    testData.forEach((element) => {
      it(`Call method with "${element.name}"`, () => {
        expect(avatar.instance().getColor(element.name)).toEqual(element.color);
      });
    });
  });
});

test('Avatar with src', () => {
  const tree = mount(
    <Avatar src="https://secure.gravatar.com/avatar/8bd2cedcb98c7a43c4f206e312568529"/>
  ).html();
  expect(tree).toMatchSnapshot();
});

test('Avatar with name', () => {
  const tree = mount(
    <Avatar name="Steven Reed"/>
  ).html();
  expect(tree).toMatchSnapshot();
});

test('Avatar with email', () => {
  const component = mount(
    <Avatar email={params.email}/>
  );
  component.setState({
    status      : 1,
    color       : '#006023',
    initial     : 'SR',
    displayName : 'Stven Reed',
    size        : 60,
    avatar      : 'http://1.gravatar.com/avatar/62615b629807a5c1f83e9d1060162003?d=https%3A%2F%2Fcdn2wp-templatemonster.netdna-ssl.com%2Fwp-content%2Fuploads%2F2016%2F10%2Fonepixel.png&r=g&s=60'
  });
  const tree = component.html();
  expect(tree).toMatchSnapshot();
});

test('Avatar with email but without gravatar', () => {
  const component = mount(
    <Avatar email={params.email}/>
  );
  component.setState({
    status: 0
  });
  const tree = component.html();
  expect(tree).toMatchSnapshot();
});

test('Avatar without anything', () => {
  const component = mount(
    <Avatar/>
  );
  component.setState({
    status: 0
  });
  const tree = component.html();
  expect(tree).toMatchSnapshot();
});

test('Avatar with set props', () => {
  const component = mount(
    <Avatar/>
  );
  component.setProps({
    name: params.name
  });

  const tree = component.html();
  expect(tree).toMatchSnapshot();
});
