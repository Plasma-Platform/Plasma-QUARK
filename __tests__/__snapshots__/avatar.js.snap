exports[`test Avatar with email 1`] = `"<div class=\"avatar \"><img alt=\"initials\" height=\"60\" width=\"60\" src=\"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHBvaW50ZXItZXZlbnRzPSJub25lIiB3aWR0aD0iNjAiIGhlaWdodD0iNjAiIHN0eWxlPSJiYWNrZ3JvdW5kLWNvbG9yOiAjMDA2MDIzIj48dGV4dCB0ZXh0LWFuY2hvcj0ibWlkZGxlIiB5PSI1MCUiIHg9IjUwJSIgZHk9IjAuMzVlbSIgc3R5bGU9ImZvbnQtZmFtaWx5OiAmcXVvdDtQVF9TYW5zJnF1b3Q7LHNhbnMtc2VyaWY7IGZvbnQtc2l6ZTogMThweCIgcG9pbnRlci1ldmVudHM9ImF1dG8iIGZpbGw9IndoaXRlIj5TUjwvdGV4dD48L3N2Zz4\" class=\"avatar__initials\"><img alt=\"avatar\" height=\"60\" width=\"60\" src=\"http://1.gravatar.com/avatar/62615b629807a5c1f83e9d1060162003?d=https%3A%2F%2Fcdn2wp-templatemonster.netdna-ssl.com%2Fwp-content%2Fuploads%2F2016%2F10%2Fonepixel.png&amp;r=g&amp;s=60\" class=\"avatar__image\"></div>"`;

exports[`test Avatar with email but without gravatar 1`] = `"<div class=\"avatar \"><img src=\"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHBvaW50ZXItZXZlbnRzPSJub25lIiB3aWR0aD0iNjAiIGhlaWdodD0iNjAiIHN0eWxlPSJiYWNrZ3JvdW5kLWNvbG9yOiAjMDA2MDIzIj48dGV4dCB0ZXh0LWFuY2hvcj0ibWlkZGxlIiB5PSI1MCUiIHg9IjUwJSIgZHk9IjAuMzVlbSIgc3R5bGU9ImZvbnQtZmFtaWx5OiAmcXVvdDtQVF9TYW5zJnF1b3Q7LHNhbnMtc2VyaWY7IGZvbnQtc2l6ZTogMThweCIgcG9pbnRlci1ldmVudHM9ImF1dG8iIGZpbGw9IndoaXRlIj5TVDwvdGV4dD48L3N2Zz4\" height=\"60\" width=\"60\" alt=\"image\"></div>"`;

exports[`test Avatar with name 1`] = `"<div class=\"avatar \"><img src=\"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHBvaW50ZXItZXZlbnRzPSJub25lIiB3aWR0aD0iNjAiIGhlaWdodD0iNjAiIHN0eWxlPSJiYWNrZ3JvdW5kLWNvbG9yOiAjMDA2MDIzIj48dGV4dCB0ZXh0LWFuY2hvcj0ibWlkZGxlIiB5PSI1MCUiIHg9IjUwJSIgZHk9IjAuMzVlbSIgc3R5bGU9ImZvbnQtZmFtaWx5OiAmcXVvdDtQVF9TYW5zJnF1b3Q7LHNhbnMtc2VyaWY7IGZvbnQtc2l6ZTogMThweCIgcG9pbnRlci1ldmVudHM9ImF1dG8iIGZpbGw9IndoaXRlIj5TUjwvdGV4dD48L3N2Zz4\" height=\"60\" width=\"60\" alt=\"image\"></div>"`;

exports[`test Avatar with set props 1`] = `"<div class=\"avatar \"><img src=\"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHBvaW50ZXItZXZlbnRzPSJub25lIiB3aWR0aD0iNjAiIGhlaWdodD0iNjAiIHN0eWxlPSJiYWNrZ3JvdW5kLWNvbG9yOiAjMDA2MDIzIj48dGV4dCB0ZXh0LWFuY2hvcj0ibWlkZGxlIiB5PSI1MCUiIHg9IjUwJSIgZHk9IjAuMzVlbSIgc3R5bGU9ImZvbnQtZmFtaWx5OiAmcXVvdDtQVF9TYW5zJnF1b3Q7LHNhbnMtc2VyaWY7IGZvbnQtc2l6ZTogMThweCIgcG9pbnRlci1ldmVudHM9ImF1dG8iIGZpbGw9IndoaXRlIj5URTwvdGV4dD48L3N2Zz4\" height=\"60\" width=\"60\" alt=\"image\"></div>"`;

exports[`test Avatar with src 1`] = `"<div class=\"avatar \"><img src=\"https://secure.gravatar.com/avatar/8bd2cedcb98c7a43c4f206e312568529\" height=\"60\" width=\"60\" alt=\"image\"></div>"`;

exports[`test Avatar without anything 1`] = `null`;
